jiant/shared/model_resolution.py

```diff
@@ -107,6 +133,16 @@ def from_encoder(cls, encoder):
             and encoder.__class__.__name__ == "XlmRobertaModel"
         ):
             return cls.XLM_ROBERTA
+        elif (
+            isinstance(encoder, transformers.BartModel)
+            and encoder.__class__.__name__ == "BartModel"
+        ):
+            return bart_or_mbart_model_heuristic(model_config=encoder.config)
```

Comment:
```suggestion
            if encoder_config.is_valid_mbart():
                return ModelArchitectures.MBART
            else:
                return ModelArchitectures.BART
```

Reply:
I'm leaning toward keeping it as is because it's a heuristic and I think it's slightly more future-proof if we keep ownership of it and wrap the underlying method.